scrapy/pipelines/files.py

```diff
@@ -248,6 +250,42 @@ def persist_file(self, path, buf, info, meta=None, headers=None):
         )
 
 
+class FTPFilesStore(object):
+    
+    def __init__(self, uri):
+        assert uri.startswith('ftp://')
+        u = urlparse(uri)
+        self.ftp = FTP()
+        self.ftp.connect(u.hostname, u.port or '21')
+        self.ftp.login(u.username, u.password)
+        self.basedir = u.path + '/'
+        ftp_makedirs_cwd(self.ftp, self.basedir+'full')
```

Comment:
Why are you modifying the received path, adding a `/` to it?

Moreover, in this line you are creating a hardcoded `/full` directory; while the default file pipelines do use such a directory, this code should support alternative file pipelines that do not.

You should probably extract the destination path from filenames on `persist_file`, and create the target folder if you have not created it before.

Reply:
I did this to avoid checking the existence of the directory every time in `persist_file`, but like you said and also mentioned by @webtekindo it should support custom paths